salt/cni/cilium-ds.yaml.jinja

```diff
@@ -0,0 +1,171 @@
+--- 
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: cilium
+  namespace: kube-system
+spec:
+  updateStrategy:
+    type: "RollingUpdate"
+    rollingUpdate:
+      # Specifies the maximum number of Pods that can be unavailable during the update process.
+      # The current default value is 1 or 100% for daemonsets; Adding an explicit value here
+      # to avoid confusion, as the default value is specific to the type (daemonset/deployment).
+      maxUnavailable: "100%"
+  template:
+    metadata:
+      labels:
+        k8s-app: cilium
+        kubernetes.io/cluster-service: "true"
+      annotations:
+        # This annotation plus the CriticalAddonsOnly toleration makes
+        # cilium to be a critical pod in the cluster, which ensures cilium
+        # gets priority scheduling.
+        # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: >-
+          [{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]
+    spec:
+      serviceAccountName: cilium
+      initContainers:
+      - name: install-cni-conf
+        image: {{ pillar['cilium']['image'] }}
+        command:
+          - /bin/sh
+          - "-c"
+          - "cp -f /etc/cni/net.d/10-cilium-cni.conf /host/etc/cni/net.d/10-cilium-cni.conf"
+        volumeMounts:
+#        - name: cilium-plugin-config
+#          mountPath: /etc/cni/net.d/
+        - name: host-cni-conf
+          mountPath: /host/etc/cni/net.d
+      - name: install-cni-bin
+        image: {{ pillar['cilium']['image'] }}
+        command:
+          - /bin/sh
+          - "-c"
+          - "cp -f /usr/lib/cni/* /host/opt/cni/bin/"
+        volumeMounts:
+        - name: host-cni-bin
+          mountPath: /host/opt/cni/bin/
+      containers:
+      - image: {{ pillar['cilium']['image'] }}
+        imagePullPolicy: IfNotPresent
+        name: cilium-agent
+        command: [ "cilium-agent" ]
+        args:
+          - "--debug=$(CILIUM_DEBUG)"
+          - "--disable-envoy-version-check"
+          - "-t"
+          - "vxlan"
+          - "--kvstore"
+          - "etcd"
+          - "--kvstore-opt"
+          - "etcd.config=/var/lib/etcd-config/etcd.config"
+          - "--disable-ipv4=$(DISABLE_IPV4)"
+#        lifecycle:
+#          postStart:
+#            exec:
+#              command:
+#                - "/cni-install.sh"
+#          preStop:
+#            exec:
+#              command:
+#                - "/cni-uninstall.sh"
```

Comment:
We should use the same `preStop` for `flannel`. Otherwise, it will leave some CNI config files that can mess things up...